solr/core/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java

```diff
@@ -160,20 +159,38 @@ public void testBasics() {
     // now delete one
     assertU(delI("5"));
 
-    long newDelsI = ((Gauge<Number>) metrics.get(delsIName)).getValue().longValue();
-    long newCumulativeDelsI = ((Meter) metrics.get(cumulativeDelsIName)).getCount();
-    assertEquals("new delsI", 1, newDelsI);
-    assertEquals("new cumulative delsI", 1, newCumulativeDelsI - cumulativeDelsI);
+    assertEquals(
+        "Did not have the expected number of pending deletes_by_id",
+        1,
+        getGaugeOpDatapoint(reader, "solr_metrics_core_update_pending_operations", "deletes_by_id")
+            .getValue(),
+        0.0);
+    assertEquals(
+        "Did not have the expected number of cumulative deletes_by_id",
+        1,
+        getGaugeOpDatapoint(
+                reader, "solr_metrics_core_update_operations_cumulative", "deletes_by_id")
```

Comment:
just a quick observation -- the "metrics" part of these names seems redundant; no?
The "solr" part is I know needed to differentiate the JVM & Jetty.